ui/radar/radar.js

```diff
@@ -0,0 +1,281 @@
+              } else if (options.PopSoundAlert && options.PopSound && options.PopVolume) {
+                let audio = new Audio(options.PopSound);
+                audio.volume = options.PopVolume;
+                audio.play();
+              }
+            } else {
+              if (options.UpdateMonsterPos) {
+                for (let i = 0; i < this.targetMonsters.length; i++) {
+                  let monster = this.targetMonsters[i];
+                  let matchTarget = r_name.match(monster['name']);
+                  if (options.SplitSameMonster)
+                    matchTarget = r_id.match(monster['id']);
+                  if (matchTarget) {
+                    monster['pos_x'] = Number(r_pos_x);
+                    monster['pos_y'] = Number(r_pos_y);
+                    monster['pos'] = new Point2D(Number(r_pos_x), -Number(r_pos_y));
+                    monster['pos_z'] = Number(r_pos_z);
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+      // update mob info from action log
+      r = e.detail.logs[i].match(/15:(.{8}):(.*?):(.*?):(.*?):(.{8}):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):(.*?):/);
+      if (r) {
+        let r_caster_id = r[1];
+        let r_caster_name = r[2];
+        let r_ability_id = r[3];
+        let r_ability_name = r[4];
+        let r_target_id = r[5];
+        let r_target_name = r[6];
+        let r_damage = r[8];
+        let damage = calc_damage(r_damage);
+        let r_target_hp = r[23];
+        for (let i = 0; i < this.targetMonsters.length; i++) {
+          let monster = this.targetMonsters[i];
+          if (monster['id'] == r_target_id) {
+            monster['battle_time'] = Date.now() / 1000;
+            monster['current_hp'] = Math.max(Number(r_target_hp) - damage, 0);
+            if (Number(r_target_hp) - 1.1 * damage <= 0 ||
+              (Number(r_target_hp) - damage) / monster['hp'] < 0.01) {
+              monster['dom'].remove();
+              this.targetMonsters.splice(i, 1);
+            }
+            break;
+          }
+        }
+      }
+      // change instances
+      r = e.detail.logs[i].match(instanceChangedRegex[lang] || instanceChangedRegex['en']);
+      if (r)
+        this.ClearTargetMonsters();
+      // removing new combatant
+      r = e.detail.logs[i].match(/04:(.{8}):Removing combatant (.*)\. (?:.*) Max HP: (\d+)\./);
+      if (r) {
+        for (let i = 0; i < this.targetMonsters.length; i++) {
+          let monster = this.targetMonsters[i];
+          if (monster['dom'] && r[1].match(monster['id']) &&
+                Date.now() / 1000 <= monster['battle_time'] + 60) {
+            monster['dom'].remove();
+            this.targetMonsters.splice(i, 1);
+            break;
+          }
+        }
+      }
+    }
+  }
+
+  OnPlayerChange(e) {
+    this.playerPos['x'] = e.detail.pos.x;
+    this.playerPos['y'] = -e.detail.pos.y;
```

Comment:
I think all y value are negated from e.detail.pos, because the y axis is upside down in the map, so, in order to doing math in a formal (high school mathematical) way, I just negated the y value.